knife/lib/chef/knife/ssh.rb

```diff
@@ -358,21 +364,22 @@ def ssh_command(command, subsession = nil)
         subsession ||= session
         command = fixup_sudo(command)
         command.force_encoding("binary") if command.respond_to?(:force_encoding)
-        begin
-          open_session(subsession, command)
-        rescue => e
-          open_session(subsession, command, true)
-        end
+        open_session(subsession, command)
       end
 
-      def open_session(subsession, command, pty = false)
+      def open_session(subsession, command)
         stderr = ""
         exit_status = 0
         subsession.open_channel do |chan|
           if config[:on_error] && exit_status != 0
             chan.close
           else
-            chan.request_pty if pty
+            if config[:ssh_pty]
+              chan.request_pty do |ch, success|
+                ui.error("Requesting PTY failed. If a PTY is not required use --no-ssh-pty") unless success
+              end
+            end
```

Comment:
Can we make the default to acquire a PTY here and then if it can't, to just debug log that the PTY wasn't acquired?

Then have an override for "--require-pty" or "--require-no-pty" to either force raising here on failure to get a pty, or to just ignore getting a pty.